### PR TITLE
Restore proper STOP/KILL supervisor mechanisms in CSIController

### PR DESCRIPTION
**Pull request by pietrzakacper**
This fixes: #258 

The implemented behavior should resemble the one from before Supervisor removal.
I guess all these flows should be revisited and refactored. They are a bit messy and quite complex.

Changes include:
* Process kill after Runner is not exiting after KILL control msg
* Send KILL msg after Runner is not emitting SEQUENCE_STOPPED after STOP control msg
* Process kill after Runner is not exiting after sending SEQUENCE_COMPLETED
* BDD test to confirm that Runner that is not responding is correctly killed

### Veryify
New CI test should confirm the case when Runner is not responding and the instance adapter removes the process forcefully after instance kill command

### CODE REVIEW

File: packages/host/src/lib/csi-controller.ts [hunk 1]
```diff
@@ -38,8 +38,9 @@ import { getRouter } from "@scramjet/api-server";
 import { getInstanceAdapter } from "@scramjet/adapters";
 import { defer, promiseTimeout, TypedEmitter } from "@scramjet/utility";
 import { ObjLogger } from "@scramjet/obj-logger";
+import { ReasonPhrases } from "http-status-codes";
 
-const stopTimeout = 7000;
+const runnerExitDelay = 11000;
```
Comment (jan-warchol) [inline packages/host/src/lib/csi-controller.ts:43]: why 11000? Is it related to the `exitDelay` in `runner.ts`?

File: packages/runner/src/runner.ts [hunk 2]
```diff
@@ -371,14 +371,16 @@ export class Runner<X extends AppConfig> implements IComponent {
         try {
             await this.runSequence(sequence, args);
 
+            this.writeMonitoringMessage([RunnerMessageCode.SEQUENCE_COMPLETED, {}]);
             this.logger.trace(`Sequence completed. Waiting ${exitDelay}ms with exit.`);
 
-            await defer(exitDelay); // @TODO: investigate why we need to wait
+            await defer(exitDelay);
```
Comment (jan-warchol) [inline packages/runner/src/runner.ts:377]: Do you know why the delay is needed?

Comment (pietrzakacper) [reply packages/runner/src/runner.ts:377]: yes, it is needed to perform API checks after sequence finish

Comment (pietrzakacper) [reply packages/runner/src/runner.ts:377]: mainly for our BDD tests

**Review by jan-warchol — APPROVED** [review]
LGTM

File: packages/host/src/lib/csi-controller.ts [hunk 3]
```diff
@@ -467,8 +457,36 @@ export class CSIController extends TypedEmitter<Events> {
         // operations
         this.router.op("post", "/_monitoring_rate", RunnerMessageCode.MONITORING_RATE, this.communicationHandler);
         this.router.op("post", "/_event", RunnerMessageCode.EVENT, this.communicationHandler);
-        this.router.op("post", "/_stop", RunnerMessageCode.STOP, this.communicationHandler);
-        this.router.op("post", "/_kill", RunnerMessageCode.KILL, this.communicationHandler);
+
+        this.router.op("post", "/_stop", async (req) => {
```
Comment (MichalCz) [inline packages/host/src/lib/csi-controller.ts:461]: @pietrzakacper can you extract this function to a class method?

File: packages/host/src/lib/csi-controller.ts [hunk 4]
```diff
@@ -467,8 +457,36 @@ export class CSIController extends TypedEmitter<Events> {
...
+            await defer(timeout);
+
+            if (!this.keepAliveRequested) {
+                await this.communicationHandler.sendControlMessage(RunnerMessageCode.KILL, {});
+            }
+
+            return {};
+        }, this.communicationHandler);
+
+        this.router.op("post", "/_kill", async () => {
```
Comment (MichalCz) [inline packages/host/src/lib/csi-controller.ts:479]: And here as well...

**Review by MichalCz — COMMENTED** [review]
@pietrzakacper take a look at the comment...

**Review by patuwwy — CHANGES_REQUESTED** [review]
please rebase and see comments from other reviewers

File: packages/host/src/lib/csi-controller.ts (hunk 4 above)
Comment (pietrzakacper) [reply packages/host/src/lib/csi-controller.ts]: fixed

File: packages/host/src/lib/csi-controller.ts (hunk 3 above)
Comment (pietrzakacper) [reply packages/host/src/lib/csi-controller.ts]: fixed

**Review by patuwwy — APPROVED** [review]

**Pull request merged**